A similar path exists: `electrum_client::serde`

I get an error while trying to install rgb-node by using the local install instructions and even when taking 
![image](https://user-images.githubusercontent.com/176074/99935546-38071d00-2d61-11eb-9beb-c78ebfe0f927.png)
the docker path:
sudo apt update
sudo apt install -y build-essential pkg-config libzmq3-dev libssl-dev libpq-dev cmake
git clone https://github.com/LNP-BP/rgb-node.git
cd rgb-node
cargo build --release

I got the error:

error[E0432]: unresolved import `serde`
  --> src/api/fungible.rs:15:5
   |
15 | use serde::{Deserialize, Serialize};
   |     ^^^^^ help: a similar path exists: `electrum_client::serde`

error[E0432]: unresolved import `serde`
  --> src/util/seal_spec.rs:16:5

Trying Docker

![image](https://user-images.githubusercontent.com/176074/99935183-63d5d300-2d60-11eb-9e6f-db344a7990f1.png)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,9 +32,9 @@ required-features = ["cli"]
 # LNP/BP crates
 amplify = "~2.3.1"
 amplify_derive = "~2.3.0"
-lnpbp = { version = "=0.2.0-beta.2", features = ["lnp", "url", "websockets"] }
-lnpbp_derive = "=0.2.0-beta.2"
-lnpbp_services = "=0.2.0-beta.2"
+lnpbp = { version = "=0.2.0-beta.3", features = ["lnp", "url", "websockets"] }
+lnpbp_derive = "=0.2.0-beta.3"
+lnpbp_services = "=0.2.0-beta.3"
 # Rust language
 lazy_static = "~1.4.0"
 nix = { version = "~0.19.0", optional = true }
@@ -104,7 +104,9 @@ shell = [
 # and other types of clients; thus `server` != `node`.
 # This feature results in building with features not required for CLI
 node = ["serde", "lnpbp/keygen", "tokio", "lnpbp/tokio", "zmq", "lnpbp_services/node",
-        "url", "lnpbp/url", "async-trait", "regex", "electrum-client", "base64"]
+        "url", "lnpbp/url", "async-trait", "regex", "electrum-client", "base64",
+        # Required for storing config and cache
+        "serde_yaml", "toml"]
 serde = ["serde_crate", "serde_with", "serde_yaml", "serde_json", "toml",
          "amplify/serde", "lnpbp/serde", "lnpbp_services/serde", "chrono/serde"]
 # Schema-specific components exposed as features:

--- a/src/api/fungible.rs
+++ b/src/api/fungible.rs
@@ -12,6 +12,7 @@
 // If not, see <https://opensource.org/licenses/MIT>.
 
 use regex::Regex;
+#[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
 use lnpbp::bitcoin::util::psbt::PartiallySignedTransaction;

--- a/src/contracts/fungible/cache/cache.rs
+++ b/src/contracts/fungible/cache/cache.rs
@@ -85,12 +85,15 @@ impl From<FileCacheError> for CacheError {
             FileCacheError::BrokenHexFilenames => Self::DataIntegrityError(
                 "Broken filename structure in storage".to_string(),
             ),
+            #[cfg(feature = "serde_json")]
             FileCacheError::SerdeJson(e) => {
                 Self::DataIntegrityError(format!("{:?}", e))
             }
+            #[cfg(feature = "serde_yaml")]
             FileCacheError::SerdeYaml(e) => {
                 Self::DataIntegrityError(format!("{:?}", e))
             }
+            #[cfg(feature = "toml")]
             FileCacheError::SerdeToml => Self::DataIntegrityError(format!(
                 "TOML serialization/deserialization error"
             )),

--- a/src/contracts/fungible/cache/file.rs
+++ b/src/contracts/fungible/cache/file.rs
@@ -11,10 +11,17 @@
 // along with this software.
 // If not, see <https://opensource.org/licenses/MIT>.
 
+#[cfg(feature = "serde")]
 use serde_json;
 use std::collections::BTreeMap;
+#[cfg(any(
+    feature = "serde_yaml",
+    feature = "serde_json",
+    feature = "toml"
+))]
+use std::io::{Read, Write};
 use std::path::PathBuf;
-use std::{fs, io, io::Read, io::Write};
+use std::{fs, io};
 
 use lnpbp::bitcoin;
 use lnpbp::rgb::prelude::*;
@@ -28,6 +35,7 @@ use crate::DataFormat;
 
 #[derive(Debug, Display, Error, From)]
 #[display(Debug)]
+#[non_exhaustive]
 pub enum FileCacheError {
     #[from]
     Io(io::Error),
@@ -41,12 +49,15 @@ pub enum FileCacheError {
     #[from]
     Encoding(lnpbp::strict_encoding::Error),
 
+    #[cfg(feature = "serde")]
     #[from]
     SerdeJson(serde_json::Error),
 
+    #[cfg(feature = "serde")]
     #[from]
     SerdeYaml(serde_yaml::Error),
 
+    #[cfg(feature = "serde")]
     #[from(toml::de::Error)]
     #[from(toml::ser::Error)]
     SerdeToml,
@@ -124,14 +135,17 @@ impl FileCache {
         let filename = self.config.assets_filename();
         let mut f = file(filename, FileMode::Read)?;
         self.assets = match self.config.data_format {
+            #[cfg(feature = "serde_yaml")]
             DataFormat::Yaml => serde_yaml::from_reader(&f)?,
+            #[cfg(feature = "serde_json")]
             DataFormat::Json => serde_json::from_reader(&f)?,
+            #[cfg(feature = "toml")]
             DataFormat::Toml => {
                 let mut data = String::new();
                 f.read_to_string(&mut data)?;
                 toml::from_str(&data)?
             }
-            DataFormat::StrictEncode => StrictDecode::strict_decode(f)?,
+            DataFormat::StrictEncode => StrictDecode::strict_decode(&mut f)?,
         };
         Ok(())
     }
@@ -142,11 +156,14 @@ impl FileCache {
         let _ = fs::remove_file(&filename);
         let mut f = file(filename, FileMode::Create)?;
         match self.config.data_format {
+            #[cfg(feature = "serde_yaml")]
             DataFormat::Yaml => serde_yaml::to_writer(&f, &self.assets)?,
+            #[cfg(feature = "serde_json")]
             DataFormat::Json => serde_json::to_writer(&f, &self.assets)?,
+            #[cfg(feature = "toml")]
             DataFormat::Toml => f.write_all(&toml::to_vec(&self.assets)?)?,
             DataFormat::StrictEncode => {
-                self.assets.strict_encode(f)?;
+                self.assets.strict_encode(&mut f)?;
             }
         }
         Ok(())
@@ -159,8 +176,11 @@ impl FileCache {
         trace!("Exporting assets information ...");
         let assets = self.assets.values().cloned().collect::<Vec<Asset>>();
         Ok(match data_format.unwrap_or(self.config.data_format) {
+            #[cfg(feature = "serde_yaml")]
             DataFormat::Yaml => serde_yaml::to_vec(&assets)?,
+            #[cfg(feature = "serde_json")]
             DataFormat::Json => serde_json::to_vec(&assets)?,
+            #[cfg(feature = "toml")]
             DataFormat::Toml => toml::to_vec(&assets)?,
             DataFormat::StrictEncode => strict_encode(&assets)?,
         })
@@ -275,7 +295,10 @@ mod test {
         // Setup a new Filecache with same Pathbuf and Json extension from above
         let filecache_config = FileCacheConfig {
             data_dir: filepath.clone(),
+            #[cfg(feature = "serde_json")]
             data_format: DataFormat::Json,
+            #[cfg(not(feature = "serde_json"))]
+            data_format: DataFormat::StrictEncode,
         };
 
         // Init new FileCache

--- a/src/contracts/fungible/config.rs
+++ b/src/contracts/fungible/config.rs
@@ -125,7 +125,10 @@ impl Default for Config {
                 .parse()
                 .expect("Error in RGB_DATA_DIR constant value"),
             cache: FUNGIBLED_CACHE.to_string(),
+            #[cfg(feature = "serde_yaml")]
             format: DataFormat::Yaml,
+            #[cfg(not(feature = "serde"))]
+            format: DataFormat::StrictEncode,
             rpc_endpoint: FUNGIBLED_RPC_ENDPOINT
                 .parse()
                 .expect("Error in FUNGIBLED_RPC_ENDPOINT constant value"),

--- a/src/contracts/fungible/data/asset.rs
+++ b/src/contracts/fungible/data/asset.rs
@@ -18,6 +18,7 @@ use std::collections::BTreeMap;
 use std::str::FromStr;
 
 use chrono::NaiveDateTime;
+#[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
 use crate::contracts::fungible::cache::models::{

--- a/src/contracts/fungible/data/outcoins.rs
+++ b/src/contracts/fungible/data/outcoins.rs
@@ -13,6 +13,7 @@
 
 use core::str::FromStr;
 use regex::Regex;
+#[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 use std::io;
 

--- a/src/contracts/fungible/processor.rs
+++ b/src/contracts/fungible/processor.rs
@@ -13,6 +13,7 @@
 
 use chrono::Utc;
 use core::convert::TryFrom;
+#[cfg(feature = "serde")]
 use serde::Deserialize;
 use std::collections::BTreeMap;
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -77,16 +77,20 @@ use std::str::FromStr;
     FromPrimitive,
     ToPrimitive,
 )]
+#[non_exhaustive]
 pub enum DataFormat {
     /// JSON
+    #[cfg(feature = "serde_json")]
     #[display("json")]
     Json,
 
     /// YAML
+    #[cfg(feature = "serde_yaml")]
     #[display("yaml")]
     Yaml,
 
     /// TOML
+    #[cfg(feature = "toml")]
     #[display("toml")]
     Toml,
 
@@ -99,8 +103,11 @@ impl_enum_strict_encoding!(DataFormat);
 impl DataFormat {
     pub fn extension(&self) -> &'static str {
         match self {
+            #[cfg(feature = "serde_yaml")]
             DataFormat::Yaml => "yaml",
+            #[cfg(feature = "serde_json")]
             DataFormat::Json => "json",
+            #[cfg(feature = "toml")]
             DataFormat::Toml => "toml",
             DataFormat::StrictEncode => "se",
         }
@@ -119,8 +126,11 @@ impl FromStr for DataFormat {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         Ok(match &s.to_lowercase() {
+            #[cfg(feature = "serde_yaml")]
             s if s.starts_with("yaml") || s.starts_with("yml") => Self::Yaml,
+            #[cfg(feature = "serde_json")]
             s if s.starts_with("json") => Self::Json,
+            #[cfg(feature = "toml")]
             s if s.starts_with("toml") => Self::Toml,
             s if s.starts_with("se")
                 || s.starts_with("dat")

--- a/src/rgbd/config.rs
+++ b/src/rgbd/config.rs
@@ -15,6 +15,7 @@ use clap::{ArgEnum, Clap};
 use lnpbp::bp;
 use std::path::PathBuf;
 
+#[cfg(feature = "serde")]
 use serde::Deserialize;
 
 use crate::constants::*;
@@ -221,7 +222,10 @@ impl Default for Config {
             stash_rpc_endpoint: STASHD_RPC_ENDPOINT.to_string(),
             stash_pub_endpoint: STASHD_PUB_ENDPOINT.to_string(),
             cache: FUNGIBLED_CACHE.to_string(),
+            #[cfg(feature = "serde_yaml")]
             format: DataFormat::Yaml,
+            #[cfg(not(feature = "serde_yaml"))]
+            format: DataFormat::StrictEncode,
             stash: STASHD_STASH.to_string(),
             index: STASHD_INDEX.to_string(),
             p2p_endpoint: STASHD_P2P_ENDPOINT.to_string(),
@@ -253,7 +257,10 @@ impl Default for Opts {
             stash_rpc_endpoint: STASHD_RPC_ENDPOINT.to_string(),
             stash_pub_endpoint: STASHD_PUB_ENDPOINT.to_string(),
             cache: FUNGIBLED_CACHE.to_string(),
+            #[cfg(feature = "serde_yaml")]
             format: DataFormat::Yaml,
+            #[cfg(not(feature = "serde_yaml"))]
+            format: DataFormat::StrictEncode,
             stash: STASHD_STASH.to_string(),
             index: STASHD_INDEX.to_string(),
             p2p_endpoint: STASHD_P2P_ENDPOINT.to_string(),

--- a/src/util/seal_spec.rs
+++ b/src/util/seal_spec.rs
@@ -13,6 +13,7 @@
 
 use core::str::FromStr;
 use regex::Regex;
+#[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 use std::io;
 

